### PR TITLE
windows fixes

### PR DESCRIPTION
**Pull request by debris**
##### fixed winows build
* `ambiguous std::array operator << ` unsigned replaced with [std::size_t](http://en.cppreference.com/w/cpp/types/size_t)
* serpent library being always included to `libweb3jsonrpc`
* missing `Boost_CHRONO` && `Boost_DATE_TIME` libraries

### CODE REVIEW

File: solc/CMakeLists.txt [hunk 1]
```diff
@@ -13,6 +13,8 @@ add_executable(${EXECUTABLE} ${SRC_LIST} ${HEADERS})
 
 target_link_libraries(${EXECUTABLE} ${Boost_FILESYSTEM_LIBRARIES})
 target_link_libraries(${EXECUTABLE} ${Boost_PROGRAM_OPTIONS_LIBRARIES})
+target_link_libraries(${EXECUTABLE} ${Boost_DATE_TIME_LIBRARIES})
```
Comment (chriseth) [inline solc/CMakeLists.txt]: Ugh. solc needs datetime and chrono?

**Comment by chriseth** [comment]
cmake allows for transitive dependencies, does it? Wouldn't it be better to state the link dependency against boost chrono and date time at the place where it is needed?

File: solc/CMakeLists.txt (hunk 1 above)
Comment (debris) [inline solc/CMakeLists.txt]: with msvc yes :/

**Comment by debris** [comment]
yes, cmake allows it. I will try to make it cleaner

**Pull request merged**